src/frontend/src/planner/relation.rs

```diff
@@ -133,19 +134,24 @@ impl Planner {
         Ok(LogicalTableFunction::new(table_function, self.ctx()).into())
     }
 
+    pub(super) fn plan_share(&mut self, share: BoundShare) -> Result<PlanRef> {
+        match self.share_cache.get(&share.name) {
```

Comment:
Using cte table name as key could be problematic in the future, as it may not be globally unique.

The current binder implementation (`cte_to_relation`) is misleading / wrong: #3174

We may need to add an intermediate layer:
* Binder: scoped cte table name -> cte id -> cte query
* Planner: cte id -> cte query

Reply:
Indeed we need a cte id instead of cte name.

Reply:
Hold on this PR until [#7212](https://github.com/risingwavelabs/risingwave/pull/7212) gets merged.